index.js

```diff
@@ -125,19 +122,14 @@ function escapeRegExp(s) {
     return s.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
 }
 
-function matchFromRegex(s, options) {
-  if (options.regExpOptions) {
-    return new RegExp(s.source, options.regExpOptions)
-  } else {
-    var flags = s.toString().replace(/\/[^\/].*\//, '')
-    // If there is no global flag then there can only be one match
-    if (flags.indexOf('g') < 0) {
-      options.limit = 1;
-    }
-    return new RegExp(s.source, flags)
+function matchFromRegex(regex, options) {
+  // If there is no global flag then there can only be one match
+  if (!regex.global) {
```

Comment:
here you should check if the global flag is within the regex itself. you can get the flags like this:

```
var flags = s.toString().replace(/\/[^\/].*\//, '')
```
then just check if there is a g.